mlir/lib/Bytecode/Reader/BytecodeReader.cpp

```diff
@@ -2121,6 +2180,11 @@ BytecodeReader::Impl::parseIRSection(ArrayRef<uint8_t> sectionData,
 LogicalResult
 BytecodeReader::Impl::parseRegions(std::vector<RegionReadState> &regionStack,
                                    RegionReadState &readState) {
+  const auto checkSectionAlignment = [&](unsigned alignment) {
+    return this->checkSectionAlignment(
+        alignment, [&](const auto &msg) { return emitError(fileLoc, msg); });
```

Comment:
One of my comment got lost somehow, I was wondering about why are the three `checkSectionAlignment` lambda using a different error reporting idiom?